plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java

```diff
@@ -530,48 +529,71 @@ public boolean isCanceled() {
+								String url = OpenShiftServerUtils.getConnectionURL(server);
 								try {
-									String conUrl = server.getAttribute(OpenShiftServerUtils.ATTR_CONNECTIONURL, (String)null);
-									if( conUrl != null ) {
-										ConnectionURL conurl2 = ConnectionURL.forURL(conUrl);
-										IConnection con = ConnectionsRegistrySingleton.getInstance().getByUrl(conurl2);
+									if( url != null ) {
+										ConnectionURL conUrl = ConnectionURL.forURL(url);
+										IConnection con = ConnectionsRegistrySingleton.getInstance().getByUrl(conUrl);
 										model.loadResources(con);
 									}
-								} catch(UnsupportedEncodingException uee) {
-									
-								} catch(MalformedURLException murle) {
-									
+								} catch(UnsupportedEncodingException | MalformedURLException e) {
+									OpenShiftUIActivator.log(IStatus.ERROR, getConnectionErrorMessage(url, server));
 								}
 								return Status.OK_STATUS;
 							}
 						})
-				.runWhenDone(
-						new Job("Setting connection, deploy project...") {
+				.runWhenDoneIf(job -> {
+							if (connection == null) {
+								container.getDisplay().syncExec(() -> 
+									new ErrorDialog(container.getShell(), "Error", 
+											NLS.bind("Could not initialize server editor {0}", server.getName()),
+											OpenShiftUIActivator.statusFactory().errorStatus(
+													getConnectionErrorMessage(OpenShiftServerUtils.getConnectionURL(server), server)),
+											IStatus.ERROR)
+										.open());
+								return false;
+							} else {
+								return true;
+							}
+						}, new Job("Setting connection, deploy project...") {
 
 							@Override
 							protected IStatus run(IProgressMonitor monitor) {
+								if (connection == null) {
+									return Status.CANCEL_STATUS;
+								}
+
 								if (deployProject != null) {
 									model.setDeployProject(deployProject);
 								}
-								if (connection != null) {
-									model.setConnection(connection);
-								}
+								model.setConnection(connection);
 								model.setService(OpenShiftServerUtils.getService(server));
 								String sourcePath = OpenShiftServerUtils.getSourcePath(server);
-								if(!StringUtils.isEmpty(sourcePath)) {
+								if (!StringUtils.isEmpty(sourcePath)) {
 									model.setSourcePath(sourcePath);
 								}
 								String podPath = OpenShiftServerUtils.getPodPath(server);
 								if(!StringUtils.isEmpty(podPath)) {
 									model.setPodPath(podPath);
 								}
+
 								model.setInitializing(false);
 								return Status.OK_STATUS;
 							}
 						})
 				.runWhenDone(new DisableAllWidgetsJob(false, container, false, busyCursor))
 				.schedule();
 	}
+	
+	private String getConnectionErrorMessage(String url, IServerAttributes server) {
+		ConnectionURL connectionUrl = ConnectionURL.safeForURL(OpenShiftServerUtils.getConnectionURL(server));
+		if (connectionUrl == null) {
+			return "Could not find OpenShift connection for server \"{0}\"";
```

Comment:
Why don't we use property files for all such strings? I've seen it almost everywhere in openshift UI plugins. There is even an existing property file: [OpenShiftUIMessages.properties](https://github.com/jbosstools/jbosstools-openshift/blob/master/plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/OpenShiftUIMessages.properties). In opinio it's more comfortable to have things like that in one place. However it's a matter of taste.